config/v1alpha1/types_insights.go

```diff
@@ -57,8 +56,29 @@ type GatherConfig struct {
 	// Run the following command to get the names of last active gatherers:
 	// "oc get insightsoperators.operator.openshift.io cluster -o json | jq '.status.gatherStatus.gatherers[].name'"
 	// An example of disabling gatherers looks like this: `disabledGatherers: ["clusterconfig/machine_configs", "workloads/workload_info"]`
+	// +kubebuilder:validation:MaxItems=100
+	// +kubebuilder:validation:items:MaxLength=256
```

Comment:
Adding these constraints are _technically_ a breaking change for users. It looks like this is a v1alpha1 API so it _should_ be okay to make this change since we only deploy v1alpha1 APIs in modes where upgrades are unsupported.

I would recommend updating the Godoc to also mention these new constraints.

While you are at it, are there any other validations that you can put in place? Do gatherers follow a specific format that you can use to ensure users aren't able to provide a string that could never evaluate to a valid gatherer name?

Reply:
I didn't know this could be a breaking change, I added these only to pass linter check.

I would not add anything else, because we are planning to drop this field in the next version.

Reply:
Removing existing fields is also a breaking change :) - I think the same logic applies with this being v1alpha1 and thus only existing in unsupported, no upgrade path, cluster modes so it should be OK to remove.

cc @JoelSpeed for a sanity check here.

Reply:
Adding limits like this to alpha types is ok, as when we go to v1, we will forget the v1alpha1 existed

If we are dropping fields between v1alpha1 and v1 that's also acceptable, since there's no upgrade path

So we can be more relaxed here than we would otherwise be normally about potentially breaking changes